src/main/java/com/example/bank/bankTransfer/transfer/history/export/XlsxTransferHistoryGenerator.java

```diff
@@ -0,0 +1,150 @@
+        autoSizeColumns(sheet);
+        createDataRows(workbook, numberOfTransferHistories, accountNumber);
+        saveWorkbookToFile(workbook, accountNumber);
+    }
+
+    private void createHeaderRow(Sheet sheet) {
+        Row headerRow = sheet.createRow(0);
+        CellStyle headerStyle = createStyleForHeader(sheet.getWorkbook());
+        IntStream.range(0, HEADER_CELL_TITLES.size()).forEach(columnIndex -> {
+            Cell cell = headerRow.createCell(columnIndex);
+            cell.setCellValue(HEADER_CELL_TITLES.get(columnIndex));
+            cell.setCellStyle(headerStyle);
+        });
+    }
+
+    private void autoSizeColumns(Sheet sheet) {
+        IntStream.range(0, HEADER_CELL_TITLES.size()).forEach(columnIndex -> {
+            sheet.autoSizeColumn(columnIndex);
+            sheet.setColumnWidth(columnIndex, sheet.getColumnWidth(columnIndex) + HORIZONTAL_PADDING);
+        });
+    }
+
+    private void createDataRows(Workbook workbook, int rowCount, String accountNumber) {
+        CellStyle dateTimeCellStyle = createDateTimeCellStyle(workbook);
+        CellStyle decimalCellStyle = createDecimalCellStyle(workbook);
+        CellStyle defaultWrapTextStyle = createDefaultWrapTextStyle(workbook);
+        List<TransferHistory> transferHistories = transferHistoryService.transferHistoryForAccountNumber(accountNumber);
+        Sheet sheet = workbook.getSheet(TRANSFER_HISTORY_SHEET_NAME);
+        for (int i = 0; i < rowCount; i++) {
+            Row row = sheet.createRow(i + 1);
+            TransferHistory history = transferHistories.get(i);
+            fillRowWithData(row, history, dateTimeCellStyle, decimalCellStyle, defaultWrapTextStyle);
+        }
+    }
+
+    private void fillRowWithData(Row row, TransferHistory history, CellStyle dateTimeCellStyle,
+                                 CellStyle decimalCellStyle, CellStyle defaultWrapTextStyle) {
+        Cell cell = null;
+        List<Object> methodsForInsertingData = createMethodsForInsertingDataIntoCells(history);
+        for (int i = 0; i < methodsForInsertingData.size(); i++) {
+            cell = row.createCell(i);
+            if (methodsForInsertingData.get(i) instanceof LocalDateTime dateTime) {
+                cell.setCellValue(dateTime);
+                cell.setCellStyle(dateTimeCellStyle);
+                row.getSheet().setColumnWidth(i, DATE_COLUMN_WIDTH);
+            } else if (methodsForInsertingData.get(i) instanceof Long number) {
+                cell.setCellValue(number);
+            } else if (methodsForInsertingData.get(i) instanceof String name) {
+                cell.setCellValue(name);
+                cell.setCellStyle(defaultWrapTextStyle);
+            } else if (methodsForInsertingData.get(i) instanceof BigDecimal bigDecimal) {
+                cell.setCellValue(bigDecimal.doubleValue());
+                cell.setCellStyle(decimalCellStyle);
+            }
+        }
+    }
```

Comment:
Raczej bym tu takie 3 metody zrobił : 

private void populateDataRows(Sheet sheet, String accountNumber) {
        List<TransferHistory> transferHistories = transferHistoryService.transferHistoryForAccountNumber(accountNumber);
        CellStyle dateTimeCellStyle = createDateTimeCellStyle(sheet.getWorkbook());
        CellStyle decimalCellStyle = createDecimalCellStyle(sheet.getWorkbook());
        CellStyle defaultWrapTextStyle = createDefaultWrapTextStyle(sheet.getWorkbook());

        for (int i = 0; i < transferHistories.size(); i++) {
            TransferHistory history = transferHistories.get(i);
            Row row = sheet.createRow(i + 1);
            fillRowWithData(row, history, dateTimeCellStyle, decimalCellStyle, defaultWrapTextStyle);
        }
    }

    private void fillRowWithData(Row row, TransferHistory history, CellStyle dateTimeCellStyle,
                                 CellStyle decimalCellStyle, CellStyle defaultWrapTextStyle) {
        List<Object> data = List.of(history.getCreatedOn(), history.getTransferType().toString(),
                history.getExternalAccountNumber(), history.getTitle(), history.getAmount(), history.getBalance());
        for (int i = 0; i < data.size(); i++) {
            Cell cell = row.createCell(i);
            Object value = data.get(i);
            setCellValueAndStyle(cell, value, dateTimeCellStyle, decimalCellStyle, defaultWrapTextStyle);
        }
    }


 private void setCellValueAndStyle(Cell cell, Object value, CellStyle dateTimeCellStyle,
                                      CellStyle decimalCellStyle, CellStyle defaultWrapTextStyle) {
        if (value instanceof LocalDateTime) {
            cell.setCellValue((LocalDateTime) value);
            cell.setCellStyle(dateTimeCellStyle);
            row.getSheet().setColumnWidth(i, DATE_COLUMN_WIDTH);
        } else if (value instanceof BigDecimal) {
            cell.setCellValue(((BigDecimal) value).doubleValue());
            cell.setCellStyle(decimalCellStyle);
        } else {
            cell.setCellValue(value.toString());
            cell.setCellStyle(defaultWrapTextStyle);
        }
    }